src/SessionSetup/ProcessLauncherWidget.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "SessionSetup/ProcessLauncherWidget.h"
+
+#include <QFileDialog>
+#include <QLineEdit>
+
+#include "OrbitBase/Logging.h"
+#include "ui_ProcessLauncherWidget.h"
+
+namespace orbit_session_setup {
+
+ProcessLauncherWidget::ProcessLauncherWidget(QWidget* parent)
+    : QWidget(parent),
+      ui(new Ui::ProcessLauncherWidget),
+      last_launched_process_or_error_(ErrorMessage()) {
+  ui->setupUi(this);
+  ui->ProcessComboBox->lineEdit()->setPlaceholderText("Process");
+  ui->WorkingDirComboBox->lineEdit()->setPlaceholderText("Working directory");
+  ui->ArgumentsComboBox->lineEdit()->setPlaceholderText("Arguments");
+  ui->ErrorLabel->setText("");
+
+  ui->gridLayout_2->setColumnStretch(0, 90);
+}
+
+ProcessLauncherWidget::~ProcessLauncherWidget() { delete ui; }
+
+void ProcessLauncherWidget::on_BrowseProcessButton_clicked() {
+  QStringList list = QFileDialog::getOpenFileNames(this, "Select an executable file...");
+  for (auto& file : list) {
+    ui->ProcessComboBox->lineEdit()->setText(file);
+    break;
+  }
+}
```

Comment:
Does it make sense to be able to select multiple things here? Especially when used with lineEdit() only the last thing is displayed.

Reply:
Yeah that was a bit weird, replaced by proper api for a single file.